File: solutions/scala/problem-00009.scala
/*
minhuang@mocha:scala]$ time scala problem-00009.scala 
31875000

real  0m1.348s
user  0m0.296s
sys 0m0.056s
*/
for (a <- 1 to 332;
     b <- a to ((1000 - a) / 2) - 1;
     c <- Some(1000 - a - b)
     if (a*a + b*b == c*c))
  println(a*b*c)

File: solutions/ruby/problem-00009.rb
#!/usr/bin/env ruby
# [minhuang@mocha:ruby]$ time ruby problem-00009.rb 
# 31875000
#
# real  0m0.085s
# user  0m0.068s
# sys 0m0.012s
n = 1000
(1..332).each do |a|
  limit = ((n - a) / 2) - 1
  (a..limit).each do |b|
    c = n - a - b
    puts a*b*c if a*a + b*b == c*c
  end
end
